CMakeLists.txt

```diff
@@ -337,9 +338,10 @@ if(NANOARROW_TESTING
                              PUBLIC $<BUILD_INTERFACE:${NANOARROW_BUILD_INCLUDE_DIR}>
                                     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/src>
                                     $<INSTALL_INTERFACE:include>)
-  target_link_libraries(nanoarrow_testing PRIVATE nlohmann_json::nlohmann_json nanoarrow)
-  target_link_libraries(nanoarrow_testing PUBLIC nanoarrow)
+  target_link_libraries(nanoarrow_testing PRIVATE nlohmann_json::nlohmann_json)
+  target_link_libraries(nanoarrow_testing PUBLIC nanoarrow nanoarrow_coverage_config)
```

Comment:
nit: you can combine these into one invocation but if you prefer it this way that's also fine.